### PR TITLE
Remove memberships when a group gets deleted

### PR DESCRIPTION
**Pull request by matthijs-va**
When a group gets deleted, its memberships should be removed as well.

### CODE REVIEW

File: og.module [hunk 1]
```diff
@@ -71,9 +71,11 @@ function og_entity_update(EntityInterface $entity) {
  * Implements hook_entity_predelete().
  */
 function og_entity_predelete(EntityInterface $entity) {
+  /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+  $membership_manager = \Drupal::service('og.membership_manager');
+
   if (Og::isGroup($entity->getEntityTypeId(), $entity->bundle())) {
-    // Register orphaned group content and user memberships for deletion, if
-    // this option has been enabled.
+    // Register orphaned group content for deletion, if this option has been enabled.
```
Comment (MPParsley) [inline og.module]:
```suggestion
    // Register orphaned group content for deletion, if this option has been
    // enabled.
```

File: og.module [hunk 2]
```diff
@@ -71,9 +71,11 @@ function og_entity_update(EntityInterface $entity) {
  * Implements hook_entity_predelete().
  */
 function og_entity_predelete(EntityInterface $entity) {
+  /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
```
Comment (MPParsley) [inline og.module]: I'd put this later on, just before the foreach.

Comment (matthijs-va) [reply og.module]: But then I would have to get it twice.

**Review by MPParsley — APPROVED** [review]

File: og.module [hunk 3]
```diff
@@ -85,11 +88,15 @@ function og_entity_predelete(EntityInterface $entity) {
     // @todo Delete user roles.
     // @see https://github.com/amitaibu/og/issues/175
     // og_delete_user_roles_by_group($entity_type, $entity);
+
```
Comment (MPParsley) [inline og.module]:
```suggestion
```
----------------------------------
 90 | WARNING | [x] There must be no blank line following an inline comment
 90 | WARNING | [ ] There must be no blank line following an inline comment

Comment (MPParsley) [inline og.module]: @matthijs-va can you remove this line? Codesniffer fails on it.
```suggestion
```

**Comment by pfrenssen** [comment]
This is already implemented, you can see this being tested in `OgDeleteOrphansTest::testDeleteOrphans()` - at the start of the test there is 1 user membership, then the group is deleted, then the cleaning up of orphans is executed, and at the end of the test it is validated that the user membership is no longer there.

We can't clean up user memberships on `hook_entity_predelete()` like is proposed in this PR because it would cause timeouts when deleting groups with thousands of members.

By default user memberships are cleaned up on cron, but we also offer an option to delete them instantly, or to let an administrator clean them up manually through the UI by launching a batch operation. We offer a plugin type for developers who have custom requirements for removing the memberships after a group is deleted (ref `OgDeleteOrphansInterface`).

If you are seeing in your website that the user memberships are not being cleaned up, then make sure your cron jobs are correctly configured, or switch to the "Simple" plugin in the OG admin UI.

**Pull request closed without merging**